### ISSUE TITLE
Ability to apply filter expressions to Querys

### ISSUE THREAD

**Issue by boostchicken** [boostchicken/spring-data-dynamodb#27]
```
	@Query(fields = "leaveDate", limit = 1, filterExpression = "contains(#field, :value)",
			expressionMappingNames = {@ExpressionAttribute(key = "#field", value = "name")},
			expressionMappingValues = {@ExpressionAttribute(key=":value", value = "projection")})
```

This allows people the ability to use filter expressions

**Comment by boostchicken** [comment]
Added https://github.com/boostchicken/spring-data-dynamodb/pull/29

**Comment by boostchicken** [comment]
Released in 5.2.4

**Comment by adarshhm6** [comment]
@Query(fields = "leaveDate", limit = 1, filterExpression = "contains(#field, :value)",
			expressionMappingNames = {@ExpressionAttribute(key = "#field", value = "name")},
			expressionMappingValues = {@ExpressionAttribute(key=":value", parameterName = "projection")})
	List<User> findByPostCode(@Param("postCode") String postCode, @Param("projection") String projection);

It allows filtering with dynamic parameters but what if the dynamic parameter passed is Non-String? 
@ExpressionAttribute has 3 fields key, value and parameterName and all fields are of type String. Suppose I want to Query for Price >= 1000 where Price field in DynamoDB table is of Type Number. In cases of Boolean and Number fields how to filter the output.
AWS CLI has option to specify the type 
{
    ":fn":{"S":"Amazon DynamoDB"},
    ":sub":{"S":"DynamoDB Thread 1"},
    ":num":{"N":"3"}
}
S for String, N for Number.
and whether filtering is supported for Collections like Price IN (100, 200).
